### PR TITLE
refactor:The plugin width is persisted using vueuse.

### PR DESCRIPTION
**Pull request by STATICHIT**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
The plugin width is persisted using vueuse.

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced management of plugin widths with a more streamlined approach using function calls for better performance.
	- Introduced storage persistence for plugin widths across sessions.
	- Added a new dependency to improve utility functions for Vue.js applications.

- **Bug Fixes**
	- Improved clarity and maintainability in the layout management process.

- **Style**
	- Added a new CSS variable for date picker width, promoting consistent styling.
	- Consolidated attributes in the `plugin-panel` component for improved readability and cleaner code structure.
	- Restructured template layouts in various components for better organization without affecting functionality.

### CODE REVIEW

**Review by lizhijie429 — COMMENTED** [review]
点击右侧高级插件面板时有报错，请处理一下。
image

**Review by lizhijie429 — COMMENTED** [review]
image
请处理一下图片中的警告。

File: packages/plugins/page/src/PageSetting.vue [hunk 1]
```diff
@@ -73,8 +73,8 @@ import http from './http.js'
 
 const { COMPONENT_NAME } = constants
 const isShow = ref(false)
-const { pluginWidth,PLUGIN_NAME } = useLayout()
-const leftMargin = computed(() => pluginWidth[PLUGIN_NAME["AppManage"]])
+const { getPluginWidth, PLUGIN_NAME } = useLayout()
+const leftMargin = computed(() => getPluginWidth(PLUGIN_NAME['AppManage']))
```
Comment (hexqi) [inline packages/plugins/page/src/PageSetting.vue]: 逻辑写到setup中，非必要不要写在外层

File: packages/plugins/page/src/PageFolderSetting.vue [hunk 2]
```diff
@@ -43,8 +43,8 @@ import PageGeneral from './PageGeneral.vue'
 import http from './http.js'
 
 let isShow = ref(false)
-const { pluginWidth, PLUGIN_NAME } = useLayout()
-const leftMargin = computed(() => pluginWidth[PLUGIN_NAME['AppManage']])
+const { getPluginWidth, PLUGIN_NAME } = useLayout()
+const leftMargin = computed(() => getPluginWidth(PLUGIN_NAME['AppManage']))
```
Comment (hexqi) [inline packages/plugins/page/src/PageFolderSetting.vue]: 需要挪到setup里面

File: packages/plugins/datasource/src/DataSourceRemotePanel.vue [hunk 3]
```diff
@@ -59,8 +59,8 @@ import { obj2String, string2Obj } from '@opentiny/tiny-engine-controller/adapter
 import { getRequest } from './js/datasource'
 
 export const isOpen = ref(false)
-const { pluginWidth, PLUGIN_NAME } = useLayout()
-const leftMargin = computed(() => pluginWidth[PLUGIN_NAME['Datasource']] - 280)
+const { getPluginWidth, PLUGIN_NAME } = useLayout()
+const leftMargin = computed(() => getPluginWidth(PLUGIN_NAME['Datasource']) - 280)
```
Comment (hexqi) [inline packages/plugins/datasource/src/DataSourceRemotePanel.vue]: 这个280是什么呢，代码中不应该出现意义不明的数字，一般是用定义成常量。
这里280看起来应该是cssvar变量，可以考虑用js读取cssvar变量值

File: packages/controller/src/useLayout.js [hunk 4]
```diff
@@ -12,6 +12,7 @@
 
 import { reactive, nextTick } from 'vue'
 import { constants } from '@opentiny/tiny-engine-utils'
+import { useStorage } from '@vueuse/core'
```
Comment (hexqi) [inline packages/controller/src/useLayout.js:15]: 依赖@vueuse/core似乎没有在package.json中声明

Comment (STATICHIT) [reply packages/controller/src/useLayout.js:15]: 有声明的呀
image

**Review by hexqi — APPROVED** [review]

**Pull request merged**